File: app/src/main/java/com/example/msg91/MainActivity.java
package com.example.msg91;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    EditText phoneNo;
    Button send;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        phoneNo = findViewById(R.id.editTextPhone);
        send = findViewById(R.id.button_send);
        send.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(),VerificationActivity.class);
                intent.putExtra("ph",phoneNo.getText().toString());
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/example/otp/MainActivity.kt
package com.example.otp

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    lateinit var phoneNo: EditText
    lateinit var send: Button
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        phoneNo = findViewById(R.id.editTextPhone)
        send = findViewById(R.id.button_send)
        send.setOnClickListener(View.OnClickListener {
            val intent = Intent(applicationContext, VerificationActivity::class.java)
            intent.putExtra("ph", phoneNo.getText().toString())
            startActivity(intent)
        })
    }
}